src/JavaCodeGenerator.cs

```diff
@@ -2515,8 +2535,12 @@ public static JavaFile GetModelJavaFile(ServiceModel model, JavaSettings setting
                         string expression = $"this.{property.Name}";
                         if (sourceTypeName == targetTypeName)
                         {
-                            if (settings.ShouldGenerateXmlSerialization && property.IsXmlWrapper)
+                            if (settings.ShouldGenerateXmlSerialization && property.IsXmlWrapper && property.WireType is ListType listType)
```

Comment:
Doesn't IsXmlWrapper imply the WireType will be a list?

Reply:
If I remember right, the main reason for this is that the pattern matching `is ListType listType` is convenient here